### PR TITLE
Gracefully handle missing xfmr modules and netlink vendor

### PR DESCRIPTION
**Pull request by aboch**
If xfrm modules cannot be loaded:
- Create netlink.Handle only for ROUTE socket
- Reject local join on overlay secure network

Requires update to vishvananda/netlink 734d02c

Also make sure `ns.Init()` is called only once in non test code (it was currently being called twice). This should solely be controlled in ns package.

Task start error would look like:
```
ERRO[0690] fatal task error                              error=cannot join secure network: required modules to install IPSEC rules are missing on host
```

### CODE REVIEW

**Comment by sanimej** [comment]
This makes the handling more graceful when xrfm module is missing but we might need more. Perhaps its not in the scope of this PR. In the cluster if some nodes are missing the xrfm module I think it will go into a non-stop loop of scheduler trying to place the task in those nodes and the task start failing. I am not sure if there is a way to indicate to the manager certain nodes  are not capable of running some tasks.

File: drivers/overlay/joinleave.go [hunk 1]
```diff
@@ -31,6 +32,12 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 		return fmt.Errorf("cannot join secure network: encryption keys not present")
 	}
 
+	nlh := ns.NlHandle()
+
+	if n.secure && !nlh.SupportsNetlinkFamily(syscall.NETLINK_XFRM) {
+		return fmt.Errorf("cannot join secure network: required modules to install IPSEC rules are missing on host")
+	}
```
Comment (sanimej) [inline drivers/overlay/joinleave.go]: It will be useful to mention the specific modules that are missing.

**Comment by aboch** [comment]
@sanimej Agree. In my testing the manager did not re-schedule the task on the another node, it kept retrying. But as you said, it should not be the goal of this PR to fix the above bahavior.

What this change primarily fixes is the ability to boot the daemon on a host which does not have the xfrm modules. As of now it would crash.

File: drivers/overlay/joinleave.go (hunk 1 above)
Comment (aboch) [inline drivers/overlay/joinleave.go]: This is already in the daemon logs, printed during `ns.init()` call as Warning.
At this point in the code, we do not know which of the two modules failed the modprobe.

**Comment by aboch** [comment]
Also consider the docker `check-config.sh` script o/p will include the required modules like:

```
- Network Drivers:
  - "overlay":
    - CONFIG_VXLAN: enabled (as module)
    Optional:
    - CONFIG_XFRM_ALGO: enabled (as module)
    - CONFIG_XFRM_USER: enabled (as module)
[...]
```

**Comment by sanimej** [comment]
LGTM

**Comment by aboch** [comment]
Reproduced on gentoo by removing  `/lib/modules/3.18.7-gentoo/kernel/net/xfrm/xfrm_user.ko`:

```
$ sudo dockerd
[...]
WARN[0001] Could not load necessary modules for IPSEC rules: Running modprobe xfrm_user failed with message: `modprobe: ERROR: could not insert 'xfrm_user': Unknown symbol in module, or unknown parameter (see dmesg)
insmod /lib/modules/3.18.7-gentoo/kernel/net/xfrm/xfrm_user.ko`, error: exit status 1 
[...]
ERRO[0072] fatal task error                              error=cannot join secure network: required modules to install IPSEC rules are missing on host
github.com/docker/docker/daemon/cluster/executor/container.(*controller).Start
	/go/src/github.com/docker/docker/daemon/cluster/executor/container/controller.go:139: starting container failed module=taskmanager task.id=34j3mwa33itl2rs9d4dpmap1c
[...]
```

**Pull request merged**